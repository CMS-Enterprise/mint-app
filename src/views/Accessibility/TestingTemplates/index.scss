.accessibility-testing-templates {
  &__table-of-contents {
    padding-left: 0;
    margin: 0;
    list-style-type: none;

    li {
      margin-bottom: 8px;

      &:last-child {
        margin-bottom: 0;
      }

      &:before {
          content: "-";
          padding-right: 8px;
      }
    }
  }

  &__vpat-list, &__test-plan-list, &__remediation-plan-list {
    padding-left: 20px;
  }

  &__sidebar {
    border-top: 4px solid $easi-bright-blue;
  }

  h2 {
    margin-top: 48px;
  }

  &__alert-note {
    i {
    font-size: 1.5em;
    }
  }
}
