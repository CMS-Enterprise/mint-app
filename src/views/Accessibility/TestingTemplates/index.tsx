import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Link as UswdsLink,
  SummaryBox
} from '@trussworks/react-uswds';

import RemediationPlanDoc from 'assets/files/CMS508RemediationPlanTemplate.pdf';
import TestPlanDoc from 'assets/files/Section508TestPlanTemplate.pdf';
import PageHeading from 'components/PageHeading';

import './index.scss';

const TestingTemplates = () => {
  const { t } = useTranslation('accessibility');
  const vpatConformanceLevels: { name: string; description: string }[] = t(
    'testingTemplates.vpatSection.subSection.item1.levels',
    {
      returnObjects: true
    }
  );
  const testPlanList: string[] = t(
    'testingTemplates.testPlanSection.itemsToProvide',
    {
      returnObjects: true
    }
  );

  const remediationPlanList: string[] = t(
    'testingTemplates.remediationPlanSection.itemsToProvide',
    {
      returnObjects: true
    }
  );

  const tableOfContents = (
    <div className="accessibility-testing-templates">
      <p className="margin-bottom-1">Page contents</p>
      <ul className="accessibility-testing-templates__table-of-contents">
        <li>
          <UswdsLink href="#vpat">
            {t('testingTemplates.vpatSection.heading')}
          </UswdsLink>
        </li>
        <li>
          <UswdsLink href="#test-plan">
            {t('testingTemplates.testPlanSection.heading')}
          </UswdsLink>
        </li>
        <li>
          <UswdsLink href="#remediation-plan">
            {t('testingTemplates.remediationPlanSection.heading')}
          </UswdsLink>
        </li>
      </ul>
    </div>
  );

  const testPlanSection = (
    <div>
      <h2 id="test-plan">{t('testingTemplates.testPlanSection.heading')}</h2>
      <p>{t('testingTemplates.testPlanSection.description')}</p>
      <ul className="accessibility-testing-templates__test-plan-list">
        {testPlanList.map(item => (
          <li key={item}>{item}</li>
        ))}
      </ul>
      <SummaryBox
        heading={t('testingTemplates.testPlanSection.download.heading')}
      >
        <UswdsLink href={TestPlanDoc} target="_blank">
          {t('testingTemplates.testPlanSection.download.link')}
        </UswdsLink>
      </SummaryBox>
    </div>
  );

  const remediationPlanSection = (
    <div>
      <h2 id="remediation-plan">
        {t('testingTemplates.remediationPlanSection.heading')}
      </h2>
      <p>{t('testingTemplates.remediationPlanSection.description')}</p>
      <ul className="accessibility-testing-templates__remediation-plan-list">
        {remediationPlanList.map(item => (
          <li key={item}>{item}</li>
        ))}
      </ul>
      <SummaryBox
        heading={t('testingTemplates.remediationPlanSection.download.heading')}
      >
        <UswdsLink href={RemediationPlanDoc} target="_blank">
          {t('testingTemplates.remediationPlanSection.download.link')}
        </UswdsLink>
      </SummaryBox>
    </div>
  );

  const downloadVPAT = (
    <SummaryBox
      heading={t(
        'testingTemplates.vpatSection.subSection.downloadVPAT.heading'
      )}
    >
      <p>
        <UswdsLink
          href="https://www.itic.org/policy/accessibility/vpat"
          target="_blank"
          rel="noopener noreferrer"
        >
          {t(
            'testingTemplates.vpatSection.subSection.downloadVPAT.line1.linkText'
          )}
        </UswdsLink>
        {` `}
        {t(
          'testingTemplates.vpatSection.subSection.downloadVPAT.line1.otherText'
        )}
      </p>
      <p className="display-flex flex-row flex-align-center accessibility-testing-templates__alert-note">
        <i className="fa fa-exclamation-circle margin-right-1" />
        {` `}
        {t('testingTemplates.vpatSection.subSection.downloadVPAT.line2.text')}
      </p>
      <p>
        <UswdsLink
          href="https://www.youtube.com/watch?v=kAkSV9xiJ1A"
          target="_blank"
          rel="noopener noreferrer"
        >
          {t(
            'testingTemplates.vpatSection.subSection.downloadVPAT.line3.linkText'
          )}
        </UswdsLink>
      </p>
    </SummaryBox>
  );

  const vpatSection = (
    <div>
      <h2 id="vpat">{t('testingTemplates.vpatSection.heading')}</h2>
      <p>{t('testingTemplates.vpatSection.description')}</p>
      <h3>{t('testingTemplates.vpatSection.subSection.heading')}</h3>
      <ul className="accessibility-testing-templates__vpat-list">
        <li>
          {t('testingTemplates.vpatSection.subSection.item1.text')}
          <div className="padding-left-2">
            <dl title="Conformance levels">
              {vpatConformanceLevels.map(level => (
                <div key={level.name}>
                  <dt className="text-bold display-inline">{level.name}</dt>{' '}
                  <dd className="margin-left-0 display-inline">
                    {level.description}
                  </dd>
                </div>
              ))}
            </dl>
          </div>
        </li>
        <li>{t('testingTemplates.vpatSection.subSection.item2.text')}</li>
      </ul>
      {downloadVPAT}
    </div>
  );

  return (
    <div className="grid-container accessibility-testing-templates">
      <BreadcrumbBar variant="wrap">
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to="/">
            <span>Home</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb current>Templates for 508 testing</Breadcrumb>
      </BreadcrumbBar>

      <div className="grid-row grid-gap-lg margin-top-6">
        <div className="grid-col-9 line-height-body-4">
          <div className="tablet:grid-col-10">
            <PageHeading className="margin-top-0">
              {t('testingTemplates.heading')}
            </PageHeading>
            {tableOfContents}
            {vpatSection}
            {testPlanSection}
            {remediationPlanSection}
          </div>
        </div>
        <div className="grid-col-3 accessibility-testing-templates__sidebar">
          <div>
            <h4>Need help? Contact the Section 508 team</h4>
            <UswdsLink href="mailto:CMS_Section508@cms.hhs.gov">
              CMS_Section508@cms.hhs.gov
            </UswdsLink>
          </div>
        </div>
      </div>
    </div>
  );
};

export default TestingTemplates;
