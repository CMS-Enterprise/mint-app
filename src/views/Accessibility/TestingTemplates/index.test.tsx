import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';

import TestingTemplates from './index';

describe('TestingTemplates', () => {
  it('renders without crashing', () => {
    const component = render(
      <MemoryRouter>
        <TestingTemplates />
      </MemoryRouter>
    );
    expect(component.length).toBe(1);
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <TestingTemplates />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
