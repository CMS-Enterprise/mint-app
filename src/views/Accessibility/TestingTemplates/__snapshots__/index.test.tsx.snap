// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestingTemplates matches the snapshot 1`] = `
<div
  className="grid-container accessibility-testing-templates"
>
  <nav
    aria-label="Breadcrumbs"
    className="usa-breadcrumb usa-breadcrumb--wrap"
  >
    <ol
      className="usa-breadcrumb__list"
    >
      <li
        className="usa-breadcrumb__list-item"
      >
        <a
          className="usa-breadcrumb__link"
          href="/"
          onClick={[Function]}
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        aria-current="page"
        className="usa-breadcrumb__list-item usa-current"
      >
        Templates for 508 testing
      </li>
    </ol>
  </nav>
  <div
    className="grid-row grid-gap-lg margin-top-6"
  >
    <div
      className="grid-col-9 line-height-body-4"
    >
      <div
        className="tablet:grid-col-10"
      >
        <h1
          aria-live="polite"
          className="easi-h1 margin-top-0"
          tabIndex={-1}
        >
          Templates for 508 testing
        </h1>
        <div
          className="accessibility-testing-templates"
        >
          <p
            className="margin-bottom-1"
          >
            Page contents
          </p>
          <ul
            className="accessibility-testing-templates__table-of-contents"
          >
            <li>
              <a
                className="usa-link"
                href="#vpat"
              >
                Voluntary Product Assessment Template (VPAT)
              </a>
            </li>
            <li>
              <a
                className="usa-link"
                href="#test-plan"
              >
                Test Plan
              </a>
            </li>
            <li>
              <a
                className="usa-link"
                href="#remediation-plan"
              >
                Remediation Plan
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h2
            id="vpat"
          >
            Voluntary Product Assessment Template (VPAT)
          </h2>
          <p>
            A VPAT is a document that helps CMS and other government agencies understand how you will meet the 508 Standards for IT accessibility.
          </p>
          <h3>
            Tips to complete a VPAT
          </h3>
          <ul
            className="accessibility-testing-templates__vpat-list"
          >
            <li>
              For each row in the table, indicate the conformance level. There are 5 levels:
              <div
                className="padding-left-2"
              >
                <dl
                  title="Conformance levels"
                >
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Supports:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The functionality of the product has at least one method that meets the criterion without known defects or meets with equivalent facilitation.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Partially Supports:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      Some functionality of the product does not meet the criterion.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Does Not Support:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The majority of product functionality does not meet the criterion.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Not Applicable:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The criterion is not relevant to the product.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Not Evaluated:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The product has not been evaluated against the criterion. This can be used only in WCAG 2.0 Level AAA.
                    </dd>
                  </div>
                </dl>
              </div>
            </li>
            <li>
              After indicating the conformance level, provide a detailed explanation in the ‘Remarks and Evaluation’ column.
            </li>
          </ul>
          <div
            className="usa-summary-box"
            data-testid="summary-box"
          >
            <div
              className="usa-summary-box__body"
            >
              <h3
                className="usa-summary-box__heading"
              >
                Download VPAT template
              </h3>
              <div
                className="usa-summary-box__text"
              >
                <p>
                  <a
                    className="usa-link"
                    href="https://www.itic.org/policy/accessibility/vpat"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Download the current VPAT template (opens in a new tab)
                  </a>
                   
                  from the Information Technology Industry Council (ITI) website.
                </p>
                <p
                  className="display-flex flex-row flex-align-center accessibility-testing-templates__alert-note"
                >
                  <i
                    className="fa fa-exclamation-circle margin-right-1"
                  />
                   
                  Do not skip any row in the document. This will result in delays.
                </p>
                <p>
                  <a
                    className="usa-link"
                    href="https://www.youtube.com/watch?v=kAkSV9xiJ1A"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Watch tutorial on how to fill out a VPAT (opens in a new tab).
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2
            id="test-plan"
          >
            Test Plan
          </h2>
          <p>
            The Test Plan is a document that helps the Section 508 team understand your system and know what areas of your system to test before it goes live. In this document you will provide:
          </p>
          <ul
            className="accessibility-testing-templates__test-plan-list"
          >
            <li>
              a description of your system
            </li>
            <li>
              information on how to set up and access it
            </li>
            <li>
              user scenarios and steps taken to perform actions
            </li>
            <li>
              and any other constraints, assumptions or risks
            </li>
          </ul>
          <div
            className="usa-summary-box"
            data-testid="summary-box"
          >
            <div
              className="usa-summary-box__body"
            >
              <h3
                className="usa-summary-box__heading"
              >
                Download Test Plan template
              </h3>
              <div
                className="usa-summary-box__text"
              >
                <a
                  className="usa-link"
                  href="Section508TestPlanTemplate.pdf"
                  target="_blank"
                >
                  Download the Test Plan template as a PDF (opens in a new tab)
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2
            id="remediation-plan"
          >
            Remediation Plan
          </h2>
          <p>
            The Remediation Plan is a document that helps the Section 508 team understand how you plan to fix the issues that were identified from the 508 test. In this document you will provide:
          </p>
          <ul
            className="accessibility-testing-templates__remediation-plan-list"
          >
            <li>
              solutions for each issue
            </li>
            <li>
              when and how you plan to schedule these changes into your project
            </li>
          </ul>
          <div
            className="usa-summary-box"
            data-testid="summary-box"
          >
            <div
              className="usa-summary-box__body"
            >
              <h3
                className="usa-summary-box__heading"
              >
                Download Remediation Plan template
              </h3>
              <div
                className="usa-summary-box__text"
              >
                <a
                  className="usa-link"
                  href="CMS508RemediationPlanTemplate.pdf"
                  target="_blank"
                >
                  Download the Remediation Plan template as a PDF (opens in a new tab)
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="grid-col-3 accessibility-testing-templates__sidebar"
    >
      <div>
        <h4>
          Need help? Contact the Section 508 team
        </h4>
        <a
          className="usa-link"
          href="mailto:CMS_Section508@cms.hhs.gov"
        >
          CMS_Section508@cms.hhs.gov
        </a>
      </div>
    </div>
  </div>
</div>
`;
